server/plugin/api.go

```diff
@@ -157,7 +157,7 @@ func (p *Plugin) withRecovery(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		defer func() {
 			if x := recover(); x != nil {
-				p.API.LogError("Recovered from a panic",
+				p.API.LogWarn("Recovered from a panic",
```

Comment:
I'm curious about this strategy in general. Do we do this in other plugins? Should we do this in all plugins?

Reply:
Are you referring to the `recover` handler? Yes, that is a pattern that all plugins should follow. Currently only a handful do that.